src/location-menu/location-menu.component.tsx

```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import { createErrorHandler } from "@openmrs/esm-error-handling";
+import {
+  getLoginLocations,
+  setSessionLocation
+} from "../location-menu/location-menu.resource";
+import style from "./location-menu.css";
+
+export default function LocationMenu(props: LocationMenuProps) {
+  const [loginLocations, setLoginLocations] = React.useState([]);
+  const [newLocation, setNewLocation]: [
+    Location,
+    (l: Location) => void
+  ] = React.useState(null);
```

Comment:
Just curious why you needed to use TypeScript typing here, but not in the previous?  I guess loginLocations is just a list, but you are wanted to special-case a Location as it's own type?

Reply:
TBH, it's just that I didn't find myself missing autocomplete for loginLocations, beyond the generic array stuff that Typescript was able to infer. It would be better to type it though :)